[24.2] Fix drag and drop for dataset collection elements

ahmedhamidawan opened the pull request:
![firefox_0MuKJwzyJh](https://github.com/user-attachments/assets/d5e5e7c4-5a37-4521-bce6-0fbcd24bbd52)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

ahmedhamidawan committed:
[24.2] Fix drag and drop for dataset collection elements

mvdbeek commented on client/src/components/Form/Elements/FormData/FormData.vue:
```diff
@@ -455,7 +447,37 @@ function canAcceptDatatype(itemDatatypes: string | Array<string>) {
+    return { historyContentType, newSrc, isADatasetElement };
+}
+
+function canAcceptSrc(element: any) {
```
There are more any's here now than necessary. Can you figure out the right type as you pull drag items out of the store ? This would make onDragEnter more robust

ahmedhamidawan committed:
improve drag and drop typing in `FormData`, fix multi drop

This adds `HistoryOrCollectionItem` typing to `FormData` and also fixes the bug with multiple item drag-drop not working.

mvdbeek reviewed (approved):
Thank you, this makes sense to me now, though I worry a little that this might get broken again in the future. A set of regression tests asserting the expected behavior would definitely be nice to have.
